improve manual entry of assignment to a data frame (#3201)
JuliaData/DataFrames.jl: docs/src/man/getting_started.md

 "M"
 "F"

julia> df[!, :C] .= 0
8-element Vector{Int64}:
 0
 0
 0
 0
 0
 0
 0
 0

julia> df
8×3 DataFrame
 Row │ A      B       C
     │ Int64  String  Int64
─────┼──────────────────────
   1 │     1  M           0
   2 │     2  F           0
   3 │     3  F           0
   4 │     4  M           0
   5 │     5  F           0
   6 │     6  M           0
   7 │     7  M           0
   8 │     8  F           0
```

The `DataFrame` we build in this way has 8 rows and 3 columns.
This can be checked using the `size` function:

```jldoctest dataframe
julia> size(df, 1)
8

julia> size(df, 2)
3

julia> size(df)
(8, 3)
```

In the above example notice that the `df[!, :C] .= 0` expression created a new

> column in a data frame by broadcasting a scalar.

Reviewer: ```suggestion
column in the data frame by broadcasting a scalar.
```

When setting a column of a data frame the `df[!, :C]` and `df.C` syntaxes are
equivalent and they would replace (or create) the `:C` column in `df`. This
is different from using `df[:, :C]` to set a column in a data frame, which
updates the contents of column in-place if it already exists.

> Here is an example showing this difference. Let us convert the `:B` column

Reviewer: ```suggestion
Here is an example showing this difference. Let us try changing the `:B` column
```

to a binary variable.

```jldoctest dataframe
julia> df[:, :B] = df.B .== "F"
ERROR: MethodError: Cannot `convert` an object of type Bool to an object of type String

julia> df[:, :B] .= df.B .== "F"
ERROR: MethodError: Cannot `convert` an object of type Bool to an object of type String
```

The above operations did not work because when you use `:` as row selector the

> `:B` column is updated in-place, and it only supports storing of strings.

Reviewer: ```suggestion
`:B` column is updated in-place, and it only supports storing strings.
```
